clients/venice-thin-client/src/main/java/com/linkedin/venice/client/store/AbstractAvroStoreClient.java

```diff
@@ -272,28 +272,36 @@ protected RecordSerializer<K> getKeySerializerForRequest() {
     }
     synchronized (this) {
       try {
+        if (keySerializer != null) {
```

Comment:
Even with this, multiple threads which crossed the check on line 269 before its set to true can call `getKeySerializerWithRetryWithShortInterval` apart from the one `asyncStoreInitThread` started by the 1st thread which infinitely retries with 1 minute sleep. Is this by design? looks okay to try that burst of init during the startup, but wanted to confirm.

Reply:
No, it is not possible to initialize the `keySerializer` multiple times since `getKeySerializerWithRetry` has a synchronized section to only allow at most one thread to enter at one time.

Reply:
yes, at most one thread can enter the synchronized block at once, but multiple threads can enter here sequentially after the first thread failed and setup `asyncStoreInitThread`.

Reply:
Synched offline with Gaojie. Multiple threads can call `getKeySerializerWithRetryWithShortInterval` sequentially during the init, but thats okay as in worst scenario: there will be more schema fetch requests at the very beginning.